test/lib/provision.sh

```
@@ -0,0 +1,216 @@
+#
+# Waits for the namespace to finish deleting before returning: the kind and
+# fresh-OCP flows reprovision immediately after teardown, and a create/apply/helm
+# against a still-`Terminating` namespace fails. Best-effort — warns rather than
+# fails if a finalizer stalls deletion past the timeout.
+teardown_cluster() {
+  local gw_name="${1:-}"
+  helm uninstall openshell -n openshell 2>/dev/null || true
+  [[ -n "$gw_name" ]] && "$CLI" gateway remove "$gw_name" 2>/dev/null || true
+  kubectl delete ns openshell --ignore-not-found --wait=false 2>/dev/null || true
+  local i
+  for i in $(seq 1 60); do
```

Shellcheck (0.11.0)

[warning] i appears unused. Verify use (or export if used externally).

(SC2034)